diff.py

```diff
@@ -2441,8 +2438,28 @@ def do_diff(lines1: List[Line], lines2: List[Line], config: Config) -> Diff:
                 if normalize_imms(branchless1, arch) == normalize_imms(
                     branchless2, arch
                 ):
-                    parts2 = branchless2.split("\t")
-                    if len(parts2) > 1 and imm_matches_everything(parts2[1], arch):
+
+                    all_fields_match = True
+                    new_parts = line2.scorable_line.split(None, 1)
+                    old_parts = line1.scorable_line.split(None, 1)
+                    newfields = new_parts[1].split(",")
+                    oldfields = old_parts[1].split(",")
+                    re_paren = re.compile(r"(?<!%hi)(?<!%lo)\(")
+                    oldfields = oldfields[:-1] + re_paren.split(oldfields[-1])
+                    newfields = newfields[:-1] + re_paren.split(newfields[-1])
+
+                    for nf, of in zip(newfields, oldfields):
+                        if nf != of:
+                            # If the new field is a match to any symbol case
+                            # and the old field had a relocation, then ignore this mismatch
+                            if (
+                                field_matches_any_symbol(nf, config.arch)
+                                and line1.has_symbol
+                            ):
+                                continue
+                            all_fields_match = False
+
```

Comment:
This is a lot of code, let's put it in a function (i.e. reinstating imm_matches_everything but perhaps with a better name)

Reply:
sounds good